### PR TITLE
Separate Jenkins Parameters into Categories 

### PR DESCRIPTION
**Pull request by patkarns**
Following the prototype from #2238

[Please see the parameters here](https://ci.adoptopenjdk.net/view/work-in-progress/job/Test_openjdk17_j9_sanity.perf_x86-64_linux/build?delay=0sec)

### CODE REVIEW

**Review by smlambert — COMMENTED** [review]
This is looking good @patkarns !  I have added a few notes, and will likely have a few more to add relating to the existing help text we have.  

We are in a code freeze right now, so won't merge until next week after our release is completed which will give others a chance to comment too.

**Review by smlambert — APPROVED** [review]
LGTM, thanks @patkarns - This is a first cut of this work a good base to add some additional links and comments in subsequent PRs.

File: buildenv/jenkins/testJobTemplate [hunk 1]
```diff
@@ -135,6 +135,14 @@ if (LIGHT_WEIGHT_CHECKOUT) {
 	SCRIPT_PATH = "buildenv/jenkins/openjdk_tests"
 }
 
+String sectionHeaderStyleCss = ' color: blue; font-family: Roboto, sans-serif !important; padding: 5px; font-weight: bold; margin-top: 0.5rem; font-size: 1rem; outline: 0; '
+
+String separatorStyleCss = ' border: 0; border-bottom: 1px solid #ccc; background: #999; '
```
Comment (llxia) [inline buildenv/jenkins/testJobTemplate:140]: I think we do not need `background: #999;`

File: buildenv/jenkins/testJobTemplate [hunk 2]
```diff
@@ -186,58 +194,169 @@ ARCH_OS_LIST.each { ARCH_OS ->
 
 					definition {
 						parameters {
+							parameterSeparatorDefinition {
+							    name('GRINDER_WIKI_LINK')
+							    separatorStyle(separatorStyleCss)
+							    sectionHeader('<p><a href="https://github.com/AdoptOpenJDK/openjdk-tests/wiki/How-to-Run-a-Grinder-Build-on-Jenkins">Grinder Wiki</a></p>')
+							    sectionHeaderStyle(' border: none; margin-top: 0; margin-bottom: 0; padding-left: 5px; ')
```
Comment (llxia) [inline buildenv/jenkins/testJobTemplate:201]: I think we can remove this one

File: buildenv/jenkins/testJobTemplate [hunk 3]
```diff
@@ -4,9 +4,9 @@
  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at
-
+ 
       https://www.apache.org/licenses/LICENSE-2.0
-
+
```
Comment (llxia) [inline buildenv/jenkins/testJobTemplate:9]: Please remove the spaces.

**Comment by llxia** [comment]
Could you please resolve the conflicts? Thanks

File: buildenv/jenkins/testJobTemplate [hunk 4]
```diff
@@ -193,58 +201,169 @@ ARCH_OS_LIST.each { ARCH_OS ->
...
+							    sectionHeader('Non-AQA Test Repositories')
+							    sectionHeaderStyle(sectionHeaderStyleCss)
+							}
+							parameterSeparatorDefinition {
+							    name('NON_AQA_TEST_REPOS_HELP_TEXT')
+							    separatorStyle(separatorHelpTextStyleCss)
+							    sectionHeader('Additional test repositories that can be overlayed onto existing AQA test material for extra testing.')
+							    sectionHeaderStyle(sectionHeaderHelpTextStyleCss)
+							}
+							stringParam('VENDOR_TEST_REPOS', "", "Optional. Addtional test repos")
```
Comment (sophia-guo) [inline buildenv/jenkins/testJobTemplate:247]: As as JDK_REPO and JDK_BRANCH, VENDOR_TEST_REPOS VENDOR_TEST_BRANCHES and VENDOR_TEST_DIRS have updated to be as parameters to be passed in.

File: buildenv/jenkins/testJobTemplate [hunk 5]
```diff
@@ -193,58 +201,169 @@ ARCH_OS_LIST.each { ARCH_OS ->
...
 							stringParam('ADOPTOPENJDK_REPO', ADOPTOPENJDK_REPO, "AdoptOpenJDK git repo. Please use ssh for zos.")
 							stringParam('ADOPTOPENJDK_BRANCH', ADOPTOPENJDK_BRANCH, "AdoptOpenJDK branch")
 							stringParam('OPENJ9_REPO', OPENJ9_REPO, "OpenJ9 git repo. Please use ssh for zos.")
 							stringParam('OPENJ9_BRANCH', "master", "OpenJ9 branch")
 							stringParam('OPENJ9_SHA', "", "OpenJ9 sha")
-							stringParam('JDK_REPO', JDK_REPO, "JDK git repo. Please use ssh for zos.")
-							stringParam('JDK_BRANCH', JDK_BRANCH, "JDK branch")
-							stringParam('JDK_VERSION', "${JDK_VERSION}", "JDK version. i.e., 8, 11")
-							stringParam('JDK_IMPL', JDK_IMPL_BASE, "JDK Base implementation, e.g. hotspot, openj9")
+							stringParam('JDK_REPO', "", "JDK git repo. Please use ssh for zos.")
```
Comment (sophia-guo) [inline buildenv/jenkins/testJobTemplate:227]: JDK_REPO, JDK_BRANCH have been updated to be parameters to  be assed in.

**Comment by llxia** [comment]
FYI https://github.com/AdoptOpenJDK/openjdk-tests/pull/2395 is updating the same file.

**Comment by smlambert** [comment]
re: https://github.com/AdoptOpenJDK/openjdk-tests/pull/2375#issuecomment-804884531, @patkarns - please resolve the conflict on this PR?

**Review by llxia — APPROVED** [review]
LGTM

**Comment by llxia** [comment]
Just for the record, tested this PR in openj9 Jenkins after OpenJ9: Add parameter-separator plugin eclipse-cbi/jiro#156 is merged. https://ci.eclipse.org/openj9/view/Sandbox/job/Test_Job_Auto_Gen_Sandbox/1/console still failed. This plugin may require restarting openj9 Jenkins.

**Comment by llxia** [comment]
Openj9 Jenkins is restarted. https://ci.eclipse.org/openj9/view/Sandbox/job/Test_Job_Auto_Gen_Sandbox/2/ passed as expected.

@patkarns Please create a new PR for this. You may need to update the PR to include the latest changes in this file. Thanks.

**Pull request merged**